python/cudf/cudf/core/index.py

```diff
@@ -439,16 +439,12 @@ def to_arrow(self):
         return self._values.to_arrow()
 
     def tolist(self):
-        """
-        Return a list type from index data.
 
-        Returns
-        -------
-        list
-        """
-        # TODO: Raise error as part
-        # of https://github.com/rapidsai/cudf/issues/5689
-        return self.to_arrow().to_pylist()
+        raise TypeError(
+            "Implicit conversion to a host memory via tolist() is not "
```

Comment:
Using `tolist()` is explicit, not implicit :) Maybe just:

```
cuDF does not support the `tolist()` method. Consider using `.to_pandas().to_list()` to construct a Python list. 
```

Reply:
Changed the message accordingly 👍